indico/modules/designer/util.py

```diff
@@ -38,3 +38,17 @@ def get_all_templates(obj):
 def get_inherited_templates(obj):
     """Get all templates inherited by a given event/category"""
     return get_all_templates(obj) - set(obj.designer_templates)
+
+
+def get_default_template_on_category(obj):
+    if isinstance(obj, Event):
+        return None
+    if obj.default_ticket_template:
+        return obj.default_ticket_template
+    if obj.is_root:
+        default_root_tpl = DesignerTemplate.find_first(DesignerTemplate.not_deletable)
```

Comment:
Isn't we weird we're doing such an assignment in a function called `get_default_template_on_category `? That's a side effect I wouldn't expect. Shouldn't we do that check outside the function, in the handler that calls it (right after deletion)?